src/code/z_camera.c

```diff
@@ -8,12 +8,24 @@ s32 Camera_ChangeModeFlags(Camera* camera, s16 mode, u8 flags);
 s32 Camera_QRegInit(void);
 s32 Camera_UpdateWater(Camera* camera);
 
-#define RELOAD_PARAMS \
-    (camera->animState == 0 || camera->animState == 0xA || camera->animState == 0x14 || R_RELOAD_CAM_PARAMS)
+#define CAM_GET_RO_DATA(type) &((type*)camera->paramData)->roData // Read-Only Data
+#define CAM_GET_RW_DATA(type) &((type*)camera->paramData)->rwData // Read-Write Data
 
-#define PCT(x) ((x)*0.01f)
-#define NEXTSETTING ((values++)->val)
-#define NEXTPCT PCT(NEXTSETTING)
+// Camera will reload its paramData. This is mainly read-only data from camera_data. Although some read-write data is
+// reset as well
+#define RELOAD_PARAMS (camera->animState == 0 || camera->animState == 10 || camera->animState == 20)
+
+/**
+ * Camera data is stored in both read-only data and OREG as s16, and then converted to the appropriate type during
+ * runtime. If a small f32 is being stored as an s16, it is common to store that value 100 times larger than the
+ * original value. This is then scaled back down during runtime with the CAM_DATA_SCALED macro.
+ */
+#define CAM_DATA_SCALED(x) ((x)*0.01f)
+
+// Load the next setting from camera_data.c
+#define GET_NEXT_RO_DATA ((values++)->val)
```

Comment:
Thoughts on passing `values` as an argument to the macro?
```suggestion
#define GET_NEXT_RO_DATA(values) ((values++)->val)
```
It's longer but imho better (as it doesn't rely on specific local names)

same with `CAM_GET_RO_DATA`/RW, and `RELOAD_PARAMS`, I guess. Except for those it wouldn't be such a big deal to make them longer I think

Reply:
For the `CAM_GET_RO_DATA`, `CAM_GET_RW_DATA`, where you thinking of something like:

`#define CAM_GET_RO_DATA(type, paramData) &((type*)paramData)->roData`

Which would look something like:

Normal1ReadOnlyData rodata = CAM_GET_RO_DATA(Normal1, camera->paramData)`

Thoughts on that? Are those reasonable arguments or could it be slightly different still?

Reply:
I was thinking passing `camera`

for the sake of comparison:

```c
#define CAM_GET_RO_DATA(type) &((type*)camera->paramData)->roData // Read-Only Data
    Normal1ReadOnlyData* roData = CAM_GET_RO_DATA(Normal1);
```

```c
#define CAM_GET_RO_DATA(type, paramData) &((type*)paramData)->roData // Read-Only Data
    Normal1ReadOnlyData* roData = CAM_GET_RO_DATA(Normal1, camera->paramData);
```

```c
#define CAM_GET_RO_DATA(type, camera) &((type*)camera->paramData)->roData // Read-Only Data
    Normal1ReadOnlyData* roData = CAM_GET_RO_DATA(Normal1, camera);
```

```c
    Normal1ReadOnlyData* roData = &((Normal1*)camera->paramData)->roData;
```

I do feel like `CAM_GET_RO_DATA(Normal1, camera)` is the best compromise between readability and abstraction but it may be really subjective and someone may come in with a different opinion

this choice would be consistent with at least `GET_PLAYER` and `ONEPOINT_CS_INFO` though 🐱